contracts/zero-ex/contracts/src/transformers/FillQuoteTransformer.sol

```diff
@@ -0,0 +1,424 @@
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibBytesV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibMathV06.sol";
+import "../errors/LibTransformERC20RichErrors.sol";
+import "../vendor/v3/IExchange.sol";
+import "./Transformer.sol";
+import "./LibERC20Transformer.sol";
+
+
+/// @dev A transformer that fills an ERC20 market sell/buy quote.
+contract FillQuoteTransformer is
+    Transformer
+{
+    /// @dev Transform data to ABI-encode and pass into `transform()`.
+    struct TransformData {
+        // The token being sold.
+        // This should be an actual token, not the ETH pseudo-token.
+        IERC20TokenV06 sellToken;
+        // The token being bought.
+        // This should be an actual token, not the ETH pseudo-token.
+        IERC20TokenV06 buyToken;
+        // The orders to fill.
+        IExchange.Order[] orders;
+        // Signatures for each respective order in `orders`.
+        bytes[] signatures;
+        // Maximum fill amount for each order. This may be shorter than the
+        // number of orders, where missing entries will be treated as `uint256(-1)`.
+        // For sells, this will be the maximum sell amount (taker asset).
+        // For buys, this will be the maximum buy amount (maker asset).
+        uint256[] maxOrderFillAmounts;
+        // Amount of `sellToken` to sell. May be `uint256(-1)` to sell entire
+        // amount of `sellToken` received. Zero if performing a market buy.
+        uint256 sellAmount;
+        // Amount of `buyToken` to buy. Zero if performing a market sell.
+        uint256 buyAmount;
+    }
+
+    /// @dev Results of a call to `_fillOrder()`.
+    struct FillOrderResults {
+        // The amount of taker tokens sold, according to balance checks.
+        uint256 takerTokenSoldAmount;
+        // The amount of maker tokens sold, according to balance checks.
+        uint256 makerTokenBoughtAmount;
+        // The amount of protocol fee paid.
+        uint256 protocolFeePaid;
+    }
+
+    /// @dev The Exchange ERC20Proxy ID.
+    bytes4 constant private ERC20_ASSET_PROXY_ID = 0xf47261b0;
+
+    /// @dev The Exchange contract.
+    IExchange public immutable exchange;
+    /// @dev The ERC20Proxy address.
+    address public immutable erc20Proxy;
+
+    using LibERC20TokenV06 for IERC20TokenV06;
```

Comment:
idk I kind of like having them down before the first function since that's when the syntactic sugar can actually get used. The rest of the code base is like this as well so I'd have to touch nearly every other contract. We can do a separate re-styling PR prior to deployment if we agree it's better.

Reply:
Okay yeah not a blocker, but we should probably put these at the top since they apply to the entire contract.